src/python_testing/TC_PAVST_2_4.py

```diff
@@ -0,0 +1,307 @@
+                4,
+                "TH2 sends the ModifyPushTransport command with ConnectionID = aConnectionID.",
+                "DUT responds with NOT_FOUND status code.",
+            ),
+            TestStep(
+                5,
+                "TH1 sends the ModifyPushTransport command with ConnectionID != aConnectionID.",
+                "DUT responds with NOT_FOUND status code.",
+            ),
+            TestStep(
+                6,
+                "TH1 Reads CurrentConnections attribute from PushAV Stream Transport Cluster on DUT over a large-payload session.",
+                "Verify the number of PushAV Connections in the list is 1. Store the TransportOptions and ConnectionID in the corresponding TransportConfiguration as aTransportOptions and aConnectionID.",
+            ),
+            TestStep(
+                7,
+                "TH1 sends the ModifyPushTransport command with ConnectionID != aConnectionID.",
+                "DUT responds with NOT_FOUND status code.",
+            ),
+        ]
+
+    @async_test_body
+    async def test_TC_PAVST_2_4(self):
+        endpoint = self.get_endpoint(default=1)
+        pvcluster = Clusters.PushAvStreamTransport
+        avcluster = Clusters.CameraAvStreamManagement
+        pvattr = Clusters.PushAvStreamTransport.Attributes
+        avattr = Clusters.CameraAvStreamManagement.Attributes
+        aSupportedFormats = []
+        aAllocatedVideoStreams = []
+        aAllocatedAudioStreams = []
+
+        aTransportOptions = ""
+        aConnectionID = ""
+
+        self.step(1)
+        # Commission DUT - already done
+        if self.pics_guard(self.check_pics("PAVST.S")):
+            transport_configs = await self.read_single_attribute_check_success(
+                endpoint=endpoint,
+                cluster=pvcluster,
+                attribute=pvattr.CurrentConnections,
+            )
+            for config in transport_configs:
+                if config.ConnectionID != 0:
+                    try:
+                        await self.send_single_cmd(
+                            cmd=pvcluster.Commands.DeallocatePushTransport(
+                                ConnectionID=config.ConnectionID
+                            ),
+                            endpoint=endpoint,
+                        )
+                    except InteractionModelError as e:
+                        asserts.assert_true(
+                            e.status == Status.Success, "Unexpected error returned"
+                        )
+
+            aSupportedFormats = await self.read_single_attribute_check_success(
+                endpoint=endpoint, cluster=pvcluster, attribute=pvattr.SupportedFormats
+            )
+            asserts.assert_greater_equal(
+                len(aSupportedFormats), 1, "SupportedFormats must not be empty!"
+            )
+
+            await self.send_single_cmd(
+                cmd=Clusters.CameraAvStreamManagement.Commands.VideoStreamAllocate(
```

Comment:
Don't hard code these values, get them from the DUT. 
See for example the stream allocation methods in TC_WEBRTCPTestBase.py